### PR TITLE
Blog

### PR DESCRIPTION
**Pull request by ozasadnyy**
## Mobile

<br/>

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -22,7 +22,10 @@
     "iron-icon": "PolymerElements/iron-icon#^2.0.0",
     "paper-button": "PolymerElements/paper-button#^2.0.0",
     "iron-lazy-pages": "TimvdLippe/iron-lazy-pages#^2.0.2",
-    "firebase": "^4.1.3"
+    "firebase": "^4.1.3",
+    "marked-element": "PolymerElements/marked-element#^2.1.1",
+    "clamp-js": "*",
```
Comment (sophieH29) [inline bower.json]: what does the "*" mean?

File: data/posts/2016-08-20-what-to-do-in-lviv.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+Ukrainian city of Lviv is no doubt one of the most charming and yet not well-known parts of Ukraine. It’s been recently gaining plenty of attention from the world, and was proudly featured on [BBC](https://www.youtube.com/watch?v=PkK93bCIofk&list=FLSdbB5_2f0i17sH1Bo-WR9A&index=1&feature=plpp_video) and [EuroNews](https://www.youtube.com/watch?v=PkK93bCIofk&list=FLSdbB5_2f0i17sH1Bo-WR9A&index=1&feature=plpp_video). City’s impressive architecture, absense of large tourist crowds and unique combination of rustiness and timelessness bring out a very special charm, capturing the hearts of everyone who’s ever visited it.
```
Comment (sophieH29) [inline data/posts/2016-08-20-what-to-do-in-lviv.md]: How about to load data from Firebase?

File: data/resources.json [hunk 3]
```diff
@@ -8,10 +8,12 @@
   "followOur": "Follow our",
   "followUs": "Follow us",
   "emailUs": "Email us",
-  "blog": "Blog",
   "partnershipResources": "Partnership Resources",
   "eventResources": "Event Resources",
   "buyTicket": "Buy ticket",
+  "footer": {
+    "blog": "Blog"
```
Comment (sophieH29) [inline data/resources.json]: Haven't you used this for the toolbar? It can be common and used in different places

File: index.html [hunk 4]
```diff
@@ -112,7 +112,7 @@
   <style>
     body {
       margin: 0;
-      font-family: 'Roboto', 'Noto', sans-serif;
+      font-family: 'Roboto', 'Helvetica Neue', Helvetica, Arial, sans-serif;;
```
Comment (sophieH29) [inline index.html]: 2 semi-colons at the end

File: src/elements/posts-list.html [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          srcset="[[post.image]]"
+          style$="background-color: [[post.backgroundColor]];"
+          sizing="cover"
+          hidden$="[[!post.image]]"
+          lazy-load preload fade
+        ></plastic-image>
+        <div flex>
+          <div class="details" layout vertical justified>
+            <div>
+              <text-truncate lines="2">
```
Comment (sophieH29) [inline src/elements/posts-list.html]: How do you think if to make this configurable, or at least move to const?

File: src/elements/shared-styles.html [hunk 6]
```diff
@@ -87,14 +90,18 @@
 
       .content {
         margin: 0 auto;
-        padding: 32px 16px;
+        padding: 24px 16px;
         max-width: 1024px;
       }
 
+      .content.content-narrow {
```
Comment (sophieH29) [inline src/elements/shared-styles.html]: maybe better just single name, .content-narrow?

File: src/pages/blog-list-page.html [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+      .image-overlay {
+        background-color: rgba(0, 0, 0, 0.6);
+      }
+
+      .details {
+        padding: 24px;
+        height: 100%;
+        transform: translateZ(0);
+        color: white;
```
Comment (sophieH29) [inline src/pages/blog-list-page.html]: #fff

File: src/pages/post-page.html [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+        if (!Object.keys(this.postsList).length) return blogActions.fetchList();
+      }
+
+      _postDataObserver(postId, postsList) {
+        if (!this.postsList || !Object.keys(this.postsList).length) return;
+        this._setHero(this.postsList[this.postData.id]);
+        this.set('source', this.postsList[this.postData.id].source);
+        this.set('published', this._getDate(this.postsList[this.postData.id].published));
+        const sortedPosts = this.transformToArray(this.postsList, 'id').sort((a, b) => b.published.localeCompare(a.published));
+        this.set('suggestedPosts', sortedPosts.filter(post => post.id !== this.postData.id).slice(0, 3));
```
Comment (sophieH29) [inline src/pages/post-page.html]: I would move lines 140-141 to separate method like getLast3Posts

File: src/pages/post-page.html [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+          "title": postInfo.title,
+          "background": {
+            "color": postInfo.primaryColor,
+            "image": postInfo.image
+          },
+          "fontColor": "#fff"
+        });
+      }
+
+      _getDate(date) {
```
Comment (sophieH29) [inline src/pages/post-page.html]: I've seen already 3 places with the same method, how about to move it to Utils Mixin?

File: bower.json (hunk 1 above)
Comment (ozasadnyy) [inline bower.json]: Any version. It's old library, so using what we have
https://libraries.io/bower/clamp-js

File: data/posts/2016-08-20-what-to-do-in-lviv.md (hunk 2 above)
Comment (ozasadnyy) [inline data/posts/2016-08-20-what-to-do-in-lviv.md]: Tried, but doesn't render well. I am thinking about storing blog posts in Firebase storage but we need the admin panel for that.
cc @unreadableusername

File: data/resources.json (hunk 3 above)
Comment (ozasadnyy) [inline data/resources.json]: Nope, there is different property, as people can rename them as they want.
Btw, please add scope to your variables

File: src/elements/posts-list.html (hunk 5 above)
Comment (ozasadnyy) [inline src/elements/posts-list.html]: Nope, it made from the design perspective. I don't want people to make it looking bad

File: src/elements/shared-styles.html (hunk 6 above)
Comment (ozasadnyy) [inline src/elements/shared-styles.html]: Will do

File: src/pages/post-page.html (hunk 8 above)
Comment (ozasadnyy) [inline src/pages/post-page.html]: Why? It's only 2 lines of code that will result in 5 after creating a method.

File: src/pages/post-page.html (hunk 9 above)
Comment (ozasadnyy) [inline src/pages/post-page.html]: Done, but I didn't commit yet

File: data/posts/2016-08-20-what-to-do-in-lviv.md (hunk 2 above)
Comment (sophieH29) [inline data/posts/2016-08-20-what-to-do-in-lviv.md]: yep, storage is a great idea

File: src/pages/post-page.html (hunk 8 above)
Comment (sophieH29) [inline src/pages/post-page.html]: Yes, but sometimes we also have to think about logical units, and the function does only 1 job, IMHO

**Pull request merged**